interfaces_assign.php UI speedup as requested on forum bounty board

loonylion opened the pull request:
Improves significantly and hopefully fixes bug#6400

Refactored interfaces_assign.php to to benefit people with large numbers of VLANs, as requested on the forum at https://forum.pfsense.org/index.php?topic=137391.0. Also contains a minor speedup for interfaces_vlan.php. Modified functions are contained in interfaces_fast.inc. Profiling code is still present but commented out, as is replaced code.

loonylion commented:
New commit also removes some commented code, some of which appeared to be slowing down page load significantly. Page load times are now much closer to generation times than previously.

sbeaver-netgate commented:
Incorrect license text in new file

loonylion commented:
licence text changed as requested

sbeaver-netgate commented on src/etc/inc/interfaces_fast.inc:
```diff
@@ -0,0 +1,271 @@
+require_once('interfaces.inc');
+/*
+* does_interface_exist_fast
+Returns an array of interrfaces which exist on the system.
```
Spelling: interrfaces

sbeaver-netgate commented on src/etc/inc/interfaces_fast.inc:
```diff
@@ -0,0 +1,271 @@
+ * convert_real_interface_to_friendly_interface_name_fast($interface): convert fxp0 -> wan, etc.
+ Returns an array of interfaces and friendly names.
+ Doesn't actually use any parameters, they're just accepted for consistency with the function it replaces
+ */
```
I don't think this consistency is required, and it rather obfuscates the code. Could these arguments be removed wherever used?

sbeaver-netgate commented on src/usr/local/www/interfaces_assign.php:
```diff
@@ -365,8 +382,13 @@ function interface_assign_description($portinfo, $portname) {
 		write_config();
+		/*$timeb = microtime(true);
+		$profile['post_write_config']=$timeb-$timea;*/
 
```
Let's just remove stuff that is no longer required.

sbeaver-netgate commented on src/usr/local/www/interfaces_assign.php:
```diff
@@ -441,18 +472,22 @@ function interface_assign_description($portinfo, $portname) {
+/*  this code scales much much better 
+0.0065770149230957 seconds
+vs
+0.49271988868713 seconds with 400 vlans*/
```
Comment not useful here

sbeaver-netgate commented on src/usr/local/www/interfaces_assign.php:
```diff
@@ -501,6 +536,32 @@ function interface_assign_description($portinfo, $portname) {
+This is too much of an improvement to ignore. Tested with 500
+VLANS, total page output is 17,197.33KB with 8456ms spent 
+waiting for page generation and transmission, compared to 
+21,697.89KB with a wait of 9734ms*/
```
This is a self-congratulatory comment.  "Keep HTML options on one line to minimize size/load-time" would suffice.

sbeaver-netgate commented on src/usr/local/www/interfaces_assign.php:
```diff
@@ -525,23 +588,31 @@ function interface_assign_description($portinfo, $portname) {
+/* replacing the port select menu generation loop that has count(interfaces) iterations 
+and is run count(interfaces) times with a pre-prepared select menu generated outside of
+this loop has produced a significant improvement in page generation and load time */
+//foreach ($portlist as $portname => $portinfo):
```
A more concise comment would be nice.

sbeaver-netgate commented on src/usr/local/www/interfaces_assign.php:
```diff
@@ -550,17 +621,23 @@ function interface_assign_description($portinfo, $portname) {
+/* As with the gettext() calls, I've removed the interface_assign_description() calls and
+replaced them with my own interface_assign_description_fast() function that's called once
+outside of the loop. Also like the gettext() edits, this change keeps paying for itself.
+ Also, removing the indents and newlines saves potentially ~9KB of HTML with 500 unassigned VLANs */
```
Comment not helpful here. "HTML indenting not used to save tmie/space" would suffice.

loonylion commented:
changes made, the long comments were intended to explain my reasoning behind some of the more significant changes or changes that appeared to possibly conflict with the coding style guide (in the case of the indent removals)

sbeaver-netgate commented:
Thanks. - Merged.
Those comments are better here than in the source I think :)

sbeaver-netgate commented:
Appears to be causing nnumerous issues: https://redmine.pfsense.org/issues/8223

sbeaver-netgate commented:
PR was reverted due to errors introduced.

loonylion commented:
Both bugs should be fixed now. I reverted back to stock pfSense code for naming OPT interfaces since the improvement wasn't huge anyway, which should take care of bug #8222 and I corrected the broken reference causing bug #8223
